contracts/src/deployers/ezeth/EzETHHyperdriveDeployerCoordinator.sol

```diff
@@ -0,0 +1,125 @@
+/// @title EzETHHyperdriveDeployerCoordinator
+/// @custom:disclaimer The language used in this code is for coding convenience
+/// @notice The deployer coordinator for the EzETHHyperdrive implementation.
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+contract EzETHHyperdriveDeployerCoordinator is HyperdriveDeployerCoordinator {
+    using FixedPointMath for uint256;
+
+    /// @notice The Renzo contract.
+    IRestakeManager public immutable restakeManager;
+
+    /// @notice The ezETH token contract.
+    IERC20 public immutable ezETH;
+
+    /// @notice Instantiates the deployer coordinator.
+    /// @param _coreDeployer The core deployer.
+    /// @param _target0Deployer The target0 deployer.
+    /// @param _target1Deployer The target1 deployer.
+    /// @param _target2Deployer The target2 deployer.
+    /// @param _target3Deployer The target3 deployer.
+    /// @param _target4Deployer The target4 deployer.
+    /// @param _restakeManager The Renzo contract.
+    constructor(
+        address _coreDeployer,
+        address _target0Deployer,
+        address _target1Deployer,
+        address _target2Deployer,
+        address _target3Deployer,
+        address _target4Deployer,
+        IRestakeManager _restakeManager
+    )
+        HyperdriveDeployerCoordinator(
+            _coreDeployer,
+            _target0Deployer,
+            _target1Deployer,
+            _target2Deployer,
+            _target3Deployer,
+            _target4Deployer
+        )
+    {
+        restakeManager = _restakeManager;
+        ezETH = IERC20(_restakeManager.ezETH());
+    }
+
+    /// @dev Prepares the coordinator for initialization by drawing funds from
+    ///      the LP, if necessary.
+    /// @param _hyperdrive The Hyperdrive instance that is being initialized.
+    /// @param _lp The LP that is initializing the pool.
+    /// @param _contribution The amount of capital to supply. The units of this
+    ///        quantity are either base or vault shares, depending on the value
+    ///        of `_options.asBase`.
+    /// @param _options The options that configure how the initialization is
+    ///        settled.
+    /// @return value The value that should be sent in the initialize
+    ///         transaction.
+    function _prepareInitialize(
+        IHyperdrive _hyperdrive,
+        address _lp,
+        uint256 _contribution,
+        IHyperdrive.Options memory _options
+    ) internal override returns (uint256 value) {
+        // Depositing as base is disallowed.
+        if (_options.asBase) {
+            revert IHyperdrive.UnsupportedToken();
+        }
+        // Otherwise, transfer vault shares from the LP and approve the
+        // Hyperdrive pool.
+        else {
+            ezETH.transferFrom(_lp, address(this), _contribution);
+            ezETH.approve(address(_hyperdrive), _contribution);
+            return value;
+        }
```

Comment:
Nit: We don't need the else.

```suggestion
        
        // Otherwise, transfer vault shares from the LP and approve the
        // Hyperdrive pool.

        ezETH.transferFrom(_lp, address(this), _contribution);
        ezETH.approve(address(_hyperdrive), _contribution);
        return value;
```